Hybrid/README.md

```diff
@@ -0,0 +1,270 @@
+    - [Create a virtual network](#vnet)
+    - [Create a public IP address](#ipaddress)
+    - [Create a virtual machine](#vm)
+    - [Stop the VM](#stop)
+    - [Start the VM](#start)
+    - [Restart the VM](#restart)
+    - [Delete the resources](#delete)
+    
+
+<a id="run"></a>
+## Run this sample
+
+1. If you don't already have it, [install Ruby and the Ruby DevKit](https://www.ruby-lang.org/en/documentation/installation/).
+
+2. If you don't have bundler, install it.
+
+    ```
+    gem install bundler
+    ```
+    
+3. Clone the repository.
+
+    ```
+    git clone https://github.com/Azure-Samples/compute-ruby-manage-vm.git
+    ```
+
+4. Install the dependencies using bundle.
+
+    ```
+    cd compute-ruby-manage-vm\Hybrid
+    bundle install
+    ```
+    
+5. Create a [service principal](https://docs.microsoft.com/en-us/azure/azure-stack/azure-stack-create-service-principals) to work against AzureStack. Make sure your service principal has [contributor/owner role](https://docs.microsoft.com/en-us/azure/azure-stack/azure-stack-create-service-principals#assign-role-to-service-principal) on your subscription.
+
+6. To authenticate the Service Principal against Azure Stack environment, the endpoints should be defined using ```get_active_directory_settings()```. Retrieve 'authenication_endpoint' and 'token_audience' from ARM metadata endpoint.
+
+    Metadata endpoint format:
+    ```
+    <ResourceManagerUrl>/metadata/endpoints?api-version=1.0
+    ```
+    Example:
+    ```
+    https://management.<location>.azurestack.external/metadata/endpoints?api-version=1.0
+    ```
+
+    ```ruby
+    def get_active_directory_settings()
+        settings = MsRestAzure::ActiveDirectoryServiceSettings.new
+        settings.authentication_endpoint = '<authentication endpoint>'
+        settings.token_audience = '<token-audience>'
+    settings
+    end
+    ```
+
+7. Set the following environment variables using the information from the service principle that you created.
+
+    ```
+    export AZURE_TENANT_ID={your tenant id}
+    export AZURE_CLIENT_ID={your client id}
+    export AZURE_CLIENT_SECRET={your client secret}
+    export AZURE_SUBSCRIPTION_ID={your subscription id}
+    ```
+
+    > [AZURE.NOTE] On Windows, use `set` instead of `export`.
+
+8. Run the sample.
+
+    ```
+    bundle exec ruby example.rb
+    ```
+
+<a id="example"></a>
+## What is example.rb doing?
+
+This sample starts by setting up ResourceManagementClient, the resource provider clients, a resource group, and a storage account using your subscription and credentials.
+
+```ruby
+  subscription_id = ENV['AZURE_SUBSCRIPTION_ID'] || '11111111-1111-1111-1111-111111111111' # your Azure Subscription Id
+  active_directory_settings = get_active_directory_settings()
+
+
+  provider = MsRestAzure::ApplicationTokenProvider.new(
+      ENV['AZURE_TENANT_ID'],
+      ENV['AZURE_CLIENT_ID'],
+      ENV['AZURE_CLIENT_SECRET'],
+      active_directory_settings
```

Comment:
Adding a comment here would help, as this active_directory_settings parameter is required only to work against azure stack or other clouds

Reply:
Comment added at Ln#36